chainer/__init__.py

```diff
@@ -89,6 +93,50 @@ def get_function_hooks():
     return ret
 
 
+def _load_array_types():
+    # Note: this function may not be protected by GIL because of external
+    # calls.
+    global _array_types
+    global _cpu_array_types
+    if _array_types is None:
+        array_types = [numpy.ndarray]
+        cpu_array_types = [numpy.ndarray]
+
+        if backends.cuda.available:
+            array_types.append(backends.cuda.ndarray)
+
+        if backends.intel64.is_ideep_available():
+            array_types.append(backends.intel64.mdarray)
+            cpu_array_types.append(backends.intel64.mdarray)
+
+        array_types = tuple(array_types)
+        cpu_array_types = tuple(cpu_array_types)
+
+        _array_types = array_types
+        _cpu_array_types = cpu_array_types
+
+
+def get_array_types():
+    _load_array_types()
+    return _array_types
+
+
+def get_cpu_array_types():
+    _load_array_types()
+    return _cpu_array_types
+
+
+def is_arrays_compatible(arrays):
+    arrays = [_ for _ in arrays if _ is not None]
+    if len(arrays) == 0:
+        return True
+    if type(arrays[0]) is backends.cuda.ndarray:
+        types = backends.cuda.ndarray
+    else:
+        types = get_cpu_array_types()
+    return all(isinstance(_, types) for _ in arrays)
```

Comment:
ditto: `_` is used as an argument of `isinstance`.

Reply:
Fixed.